openedx/openedx-platform
# Shibboleth Support (WIP)

WIP

Status: Working. Only supports "TestShib" out of the box. No UX changes yet so it works best with a limited number of providers.
### Test Instructions:
1. For devstack: As the `vagrant` user, do `sudo apt-get install libxmlsec1-dev swig`
2. Check out this branch and do `paver install_prereqs`
3. Add something like this in `~/lms.env.json`, changing the "Entity ID" to a URL related to you (doesn't need to be a real URL - e.g. "saml.yourname.com" is good)
   
   ``` json
   "SOCIAL_AUTH_TPA_SAML_SP_ENTITY_ID": "http://saml-edx-tpa.example.com/",
   "SOCIAL_AUTH_TPA_SAML_ORG_INFO": {
       "en-US": {"name": "example", "displayname": "Example Inc.", "url": "http://www.example.com"}
   }
   ```
   
   **Also** in the `FEATURES` section, make sure that you add `"ENABLE_THIRD_PARTY_AUTH": true,`
4. Generate a keypair with `openssl req -new -x509 -days 3652 -nodes -out saml.crt -keyout saml.key`
   And something like this in `~/lms.auth.json`, chaning the "MII" part for the new keys you just made:  
   
   ``` json
   "THIRD_PARTY_AUTH": {
       "TestShib A": {}
   },
   "SOCIAL_AUTH_TPA_SAML_SP_PUBLIC_CERT": "MII.....",
   "SOCIAL_AUTH_TPA_SAML_SP_PRIVATE_KEY": "MII.....",
   ```
5. Do `paver update_db`
6. Run the LMS
7. Test that the metadata is working: go to http://localhost:8000/auth/saml/metadata.xml
8. Save that XML file to your desktop and rename it to **yourname-metadata.xml**
9. Upload your metadata to https://www.testshib.org/register.html
10. Open http://localhost:8000/login **in an incognito window** and test out the "Log in with TestShib A" button.
